### ISSUE TITLE
Create CA Roo to signe the Itermediate CAs

### ISSUE THREAD

**Issue by antoniobaiao** [hyperledger-labs/minifabric#213]
Hello,

It is possible to create a CA Root self signed (or use a external one) to be the source of trust to the intermediates CAs (the organization's CAs that will signe the ententies like peer, orders etc) , instead of the Organization CAs be self signed?

The goal is to create a chain trust of CAs like in the following example link:
https://medium.com/ibm-garage/using-3rd-party-root-cas-in-hyperledger-fabric-3cafa91d1260

Thank you

**Comment by litong01** [comment]
Minifabric use OpenSSL to create certs. Then sets up ca node. You can look at the code and see how it works. I am pretty sure you can add some logic there to support using existing certs or creating new self signed certs.

**Comment by itaru2622** [comment]
This matter seems to be achieved by changing playbooks/ops/certgen/orgkeygen.yaml,
and also partykeygen.yaml (?)

**Comment by litong01** [comment]
Right. It will be somewhere there. But needs logic to import an existing cert/key from a well known ca, given that cert is allowed be intermediate ca, etc.

**Comment by antoniobaiao** [comment]
Thank all for your answers. 
I saw that "minifab" file is creating a docker container which will be used to configure all the nework loading an existent minifab image. 

It's the main.sh the begin? If yes can I run just the main.sh to do local tests?  
If there is another routine that should be done let me know please.

Thanks

**Comment by litong01** [comment]
@antoniobaiao yes, it is, but there is no good reason why you have to run it natively. it basically runs a lot of ansible playbooks inside a container, if you want to run it natively, your system will need ansible and possibly other dependencies. If you managed to get all the dependencies available in your system, you can run it without any problems.

**Comment by antoniobaiao** [comment]
Yes, you are right,
So the best way is to create a local image of minifab to be loaded on the docker container.

**Comment by litong01** [comment]
@antoniobaiao that is right. run `docker build -t hyperledgerlabs/minifab:latest .` to build your own image and run the same command, you wont have to pollute your env.

**Comment by antoniobaiao** [comment]
Thank you for your answer was really helpful. 
I already can create the root ca, however, I'm facing up some errors, those are not too explicit at the command line. In the ansible.log at /var directory, the information is very poor, so my question is, there is another place where I can eventually look at logs of ansible tasks that are being executed?

**Comment by litong01** [comment]
@antoniobaiao use -f default to see more verbose output.

**Comment by antoniobaiao** [comment]
Hello, 
Firstly thank you very much for your help.

I'm facing up some problems regarding the certificates. I add the following scripts to generate the certificates. All the ./minifab up process goes well till the creation of the channel when I receive an error. 

![image](https://user-images.githubusercontent.com/16519418/124002488-b39f0b80-d9cd-11eb-9dbf-5636a840d7bd.png)

and when I look into the container.

![image](https://user-images.githubusercontent.com/16519418/124002408-9cf8b480-d9cd-11eb-9f20-9bc9cea2989f.png)

The changes I did:

[orgkeygen.txt](https://github.com/hyperledger-labs/minifabric/files/6742759/orgkeygen.txt)
[createcerts.txt](https://github.com/hyperledger-labs/minifabric/files/6742760/createcerts.txt)

Sorry, I had to put the files in txt, github don't allow me to upload yaml files.
Can you take a look, please? 
I already check the certificates generated and it looks good, 
I used that commands here to check it 
https://www.ssl247.co.uk/kb/ssl-certificates/troubleshooting/certificate-matches-private-key

Thank you very much for your help
